### PR TITLE
{AppService} Add custom location sdk

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**
Add custom location SDK which will be used in appservice-kube extension. The commands from the appservice-kube extension will eventually be moved to core

The ExtendedLocation model is included in appservice package, however I want to be able to use ExtendedLocation management commands (like get & list: https://docs.microsoft.com/en-us/python/api/azure-mgmt-extendedlocation/azure.mgmt.extendedlocation.v2021_03_15_preview.operations.customlocationsoperations?view=azure-python-preview) in order to verify the ExtendedLocation before using it in our AppService commands

**Testing Guide**
No changes to existing commands, just adding custom location SDK 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

**Review by panchagnula — APPROVED** [review]
LGTM

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -210,7 +211,8 @@ def default_api_version(self):
         ResourceType.MGMT_APPSERVICE: '2020-09-01',
         ResourceType.MGMT_IOTHUB: '2021-03-31',
         ResourceType.MGMT_ARO: '2020-04-30',
-        ResourceType.MGMT_DATABOXEDGE: '2019-08-01'
+        ResourceType.MGMT_DATABOXEDGE: '2019-08-01',
+        ResourceType.MGMT_CUSTOMLOCATION: '2021-03-15-preview'
```
Comment (Juliehzl) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:215]: do we need to support other profiles, e.g. 2020-09-01-hybrid?

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 2]
```diff
@@ -46,6 +46,7 @@ azure-mgmt-devtestlabs==4.0.0
 azure-mgmt-dns==8.0.0
 azure-mgmt-eventgrid==9.0.0
 azure-mgmt-eventhub==4.1.0
+azure-mgmt-extendedlocation==1.0.0b2
```
Comment (Juliehzl) [inline src/azure-cli/requirements.py3.Darwin.txt:49]: is this package for app service extended location? For storage, compute and network, we support extended location in their own package like azure-mgmt-storage, azure-mgmt-compute and azure-mgmt-network. I am curious about app service design.

Comment (calvinsID) [reply src/azure-cli/requirements.py3.Darwin.txt:49]: I want to be able to use these management operations so I can verify the ExtendedLocation resource before using it in App Service: https://docs.microsoft.com/en-us/python/api/azure-mgmt-extendedlocation/azure.mgmt.extendedlocation.v2021_03_15_preview.operations.customlocationsoperations?view=azure-python-preview
We have the ExtendedLocation model defined in App Service, but don't have these management operations - does storage/compute/network verify the ExtendedLocation resources before using it?

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 1 above)
Comment (calvinsID) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:215]: I only need to use the latest, our use case is doing a GET call to ExtendedLocation to make sure the resource exists, before using it in AppService

File: src/azure-cli/requirements.py3.Darwin.txt (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/requirements.py3.Darwin.txt:49]: No, we don't.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**